src/cogent3/app/composable.py

```diff
@@ -853,18 +853,18 @@ def _disconnect(self):
 
 def _add(self, other):
     if other.input is not None:
-        raise TypeError(
+        raise ValueError(
             f"{other.__class__.__name__} already part of composed function, use disconnect() to free them up"
         )
-    elif other is self:
-        raise TypeError("cannot add an app to itself")
+    if other is self:
+        raise ValueError("cannot add an app to itself")
+
     # Check order
-    elif isinstance(self, user_function) or isinstance(other, user_function):
-        pass
-    elif self.app_type is WRITER:
+    if self.app_type is WRITER:
         raise TypeError("Left hand side of add operator must not be of type writer")
     elif other.app_type is LOADER:
         raise TypeError("Right hand side of add operator must not be of type loader")
+
     if self._return_types & {"SerialisableType", "IdentifierType"}:
```

Comment:
Can we write all of these as `elif` to be faster?

Reply:
I know they are logically unrelated, but all of them raise an error, so chaining them is possible.